modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java

```diff
@@ -126,6 +132,16 @@ public ListTypeEntry updateListTypeEntry(
 		ListTypeEntry listTypeEntry = listTypeEntryPersistence.findByPrimaryKey(
 			listTypeEntryId);
 
+		if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-169559")) &&
+			Validator.isNotNull(externalReferenceCode)) {
+
+			_validateExternalReferenceCode(
+				externalReferenceCode, listTypeEntry.getCompanyId(),
+				listTypeEntryId);
+
+			listTypeEntry.setExternalReferenceCode(externalReferenceCode);
+		}
+
 		if (Validator.isNotNull(externalReferenceCode)) {
 			listTypeEntry.setExternalReferenceCode(externalReferenceCode);
 		}
```

Comment:
I think we can delete this condition because it is includes in the previous condition.